packages/libs/deer/desert/src/token.rs

```diff
@@ -3,69 +3,202 @@ use core::fmt::{Debug, Display, Formatter};
 use deer::Number;
 
 // TODO: test
-#[derive(Debug, Copy, Clone, PartialEq)]
+// TODO: this should be `Copy`, but `Number` has no &'static constructor
+#[derive(Debug, Clone, PartialEq)]
 pub enum Token {
-    /// A serialized `bool`
+    /// A serialized [`bool`]
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&true, &[Token::Bool(true)])
+    /// ```
+    Bool(bool),
+
+    /// A serialized [`Number`]
+    ///
+    /// ```
+    /// use deer::Number;
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&Number::from(1), &[Token::Number(Number::from(1))])
+    /// ```
+    Number(Number),
+
+    /// A serialized [`u128`]
+    ///
+    /// this is a separate variant, as [`Number`] is unable to fully represent 128 bit values, this
+    /// means that u128/i128 need special support.
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&1u128, &[Token::U128(1)])
+    /// ```
+    U128(u128),
+
+    /// A serialized [`i128`]
+    ///
+    /// this is a separate variant, as [`Number`] is unable to fully represent 128 bit values, this
+    /// means that u128/i128 need special support.
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&1i128, &[Token::I128(1)])
+    /// ```
+    I128(i128),
+
+    /// A serialized [`usize`]
+    ///
+    /// this is a separate variant, because there is no guarantee about the width of a usize/isize,
+    /// it depends on the size to reference a memory address on the host system. There might be
+    /// systems that use 128 bit in the future that rusts supports, which means we too need to
```

Comment:
```suggestion
    /// systems that use 128 bit in the future that rust supports, which means we too need to
```
unless you meant to write `rustc` :)